Workspace/src/ik/src/Roboarm.cpp

```diff
@@ -53,6 +58,31 @@ bool Roboarm::calculateVelocities(double endEffector[3], double angles[3],
   double l2 = linkLengths[1];
   double l3 = linkLengths[2];
 
+  Matrix3d J;
+
+  J << -(l1 * s1 + l2 * s12 + l3 * s123), -(l2 * s12 + l3 * s123), -l3 * s123,
+      (l1 * c1 + l2 * c12 + l3 * c123), (l2 * c12 + l3 * c123), l3 * c123, 1, 1,
+      1;
+
+  Vector3d desVel;
+  desVel << eeX, eeY, eePhi;
+
+  Vector3d jointVel = J.colPivHouseholderQr().solve(desVel);
+
+  if (!(J * jointVel).isApprox(desVel, 0.01)) {
```

Comment:
Could u add a short comment about why the velocities get set to 0 here

Reply:
This eigen solver doesn't crash if the matrix is non-invertible, but requires this check for a valid solution. If the solution isnt valid then I assume the motion isnt attainable and set velocities to zero

Reply:
Maybe add some console output so we know when that happens

Reply:
I think I'll remove all output from the ik so the console doesn't get spammed. It returns false for all of these failure cases, so the arm interface can output to the ros/gui console if we want

Reply:
Okay, arm interface should output to GUI then